circus-train-housekeeping/src/main/java/com/hotels/bdp/circustrain/housekeeping/audit/CircusTrainAuditRevision.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (C) 2016-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.circustrain.housekeeping.audit;
+
+import javax.persistence.Entity;
+import javax.persistence.Table;
+
+import org.hibernate.envers.RevisionEntity;
+
+import com.hotels.housekeeping.audit.EntityAuditRevision;
+
+@Entity
+@RevisionEntity
+@Table(schema = "circus_train", name = "audit_revision")
```

Comment:
OK, so this is how we're allowing people to define the schema name. What happens if we made the parent class non-abstract with a @Table definition on it with a schema called say "housekeeping"? Would we still be able to do what we're doing here to change the name of the table? That way we can really provide a default implementation of housekeeping that "just works" out of the box but still allow customisations like this.

Reply:
Unfortunately you cannot do that

From the docs:

Mapped Superclasses:
Entities may inherit from superclasses that contain persistent state and mapping information but are not entities. That is, the superclass is not decorated with the @Entity annotation and is not mapped as an entity by the Java Persistence provider. These superclasses are most often used when you have state and mapping information common to multiple entity classes.

Mapped superclasses cannot be queried and can’t be used in EntityManager or Query operations. You must use entity subclasses of the mapped superclass in EntityManager or Query operations. Mapped superclasses can’t be targets of entity relationships. Mapped superclasses can be abstract or concrete.

Mapped superclasses do not have any corresponding tables in the underlying datastore. Entities that inherit from the mapped superclass define the table mappings.